consensus/src/messages/handlers.rs

```diff
@@ -465,7 +463,7 @@ fn prove_transaction(
 
         let block = blockchain
             .chain_store
-            .get_block_at(election_head, false, None)
+            .get_block_at(proving_block_number, false, None)
```

Comment:
This one is also important, but I just found it as a drive-by. It's was probably a copy-paste oversight.

Reply:
Yes indeed, we need to use `proving_block_number`